server/src/main/java/org/opensearch/action/admin/indices/view/ViewService.java

```diff
@@ -0,0 +1,171 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.indices.view;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ResourceNotFoundException;
+import org.opensearch.action.search.SearchAction;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.ClusterStateUpdateTask;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.View;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.action.ActionListener;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.LongSupplier;
+import java.util.stream.Collectors;
+
+/** Service to interact with views, create, retrieve, update, and delete */
+@ExperimentalApi
+public class ViewService {
+
+    private final static Logger LOG = LogManager.getLogger(ViewService.class);
+    private final ClusterService clusterService;
+    private final NodeClient client;
+    private LongSupplier timeProvider;
+
+    public ViewService(final ClusterService clusterService, final NodeClient client, final LongSupplier timeProvider) {
+        this.clusterService = clusterService;
+        this.client = client;
+        this.timeProvider = Optional.ofNullable(timeProvider).orElse(System::currentTimeMillis);
+    }
+
+    public void createView(final CreateViewAction.Request request, final ActionListener<GetViewAction.Response> listener) {
+        final long currentTime = timeProvider.getAsLong();
+
+        final List<View.Target> targets = request.getTargets()
+            .stream()
+            .map(target -> new View.Target(target.getIndexPattern()))
+            .collect(Collectors.toList());
+        final View view = new View(request.getName(), request.getDescription(), currentTime, currentTime, targets);
+
+        createOrUpdateView(Operation.CreateView, view, listener);
+    }
+
+    public void updateView(final CreateViewAction.Request request, final ActionListener<GetViewAction.Response> listener) {
+        final View originalView = getViewOrThrowException(request.getName());
+
+        final long currentTime = timeProvider.getAsLong();
+        final List<View.Target> targets = request.getTargets()
+            .stream()
+            .map(target -> new View.Target(target.getIndexPattern()))
+            .collect(Collectors.toList());
+        final View updatedView = new View(request.getName(), request.getDescription(), originalView.getCreatedAt(), currentTime, targets);
+
+        createOrUpdateView(Operation.UpdateView, updatedView, listener);
+    }
+
+    @SuppressWarnings("deprecation")
```

Comment:
What's being suppressed? (Well, deprecation warnings, obviously, but what's deprecated in this method?)

Reply:
I've removed it it wasn't impacting the build, the class `AcknowledgedResponse` is from a namespace with non-enclusive termniology.